### PR TITLE
feat: explain command

### PR DESCRIPTION
**Pull request by a-frantz**
All the functionality changes here are quite straightforward. I added a new dependency, `colored`, in order to get some "prettier" explain output. Maybe there's more we can do in that direction?

### CODE REVIEW

File: src/commands/explain.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+pub struct Args {
+    /// The rule name to explain.
+    #[arg(required = true)]
+    pub rule_name: String,
+}
+
+pub fn explain(args: Args) -> anyhow::Result<()> {
+    let name = args.rule_name;
+
+    let rules = grammar_lint::rules()
```
Comment (adthrasher) [inline src/commands/explain.rs]: It seems odd to me that we're iterating over both lists twice. Can this be streamlined?

**Review by peterhuene — APPROVED** [review]

**Review by adthrasher — APPROVED** [review]
Looks good, though I'm hoping Peter has a clever solution to avoid the two `match` statements.

**Comment by peterhuene** [comment]
I think we can live with the two `match` statements for now, as I don't think we can chain the two iterators together since they're different concrete types due to the `E` type parameter.

The solution will be to have a single list of lint rules moving forward in the new implementation, where everything is based off the AST.

I'm hoping that we might even put them in their own crate, such as `wdl-lint`.

File: README.md [hunk 2]
```diff
@@ -34,7 +34,8 @@
 
 ## 🎨 Features
 
-* **`sprocket lint`.** Lint Workflow Description Language files.
+* **`sprocket lint`** Lint Workflow Description Language files.
+* **`sprocket explain`** Explain lint warnings.
```
Comment (adthrasher) [inline README.md]: Should this be `rules` instead of `warnings`?

Comment (a-frantz) [reply README.md]: Nice catch, you are correct

**Pull request merged**